src/geodata2.erl

```diff
@@ -33,36 +43,57 @@ start_link(Name) ->
 stop() ->
     application:stop(geodata2).
 
-new(File) ->
-    case filelib:is_file(File) of
-        true ->
-            {ok, RawData} = file:read_file(File),
-            Data =
-                case is_compressed(File) of
-                    true ->
-                        zlib:gunzip(RawData);
-                    false ->
-                        RawData
-                end,
-            {ok, Meta} = geodata2_format:meta(Data),
-            %% @TODO [RTI-8302] This one could be removed after the IPv4+IPv6 MMDB is definitely used
-            set_is_ipv6_mmdb(Meta),
-            ets:new(?GEODATA2_STATE_TID, [set, protected, named_table, {read_concurrency, true}]),
-            ets:insert(?GEODATA2_STATE_TID, {data, Data}),
-            ets:insert(?GEODATA2_STATE_TID, {meta, Meta}),
-            {ok, #state{}};
-        _ ->
-            {stop, {geodata2_dbfile_not_found, File}}
-    end.
+new(MagellanFile, DomainFile) ->
+    MagellanState = case filelib:is_file(MagellanFile) of
+      true ->
+          {ok, RawData} = file:read_file(MagellanFile),
+          Data = case is_compressed(MagellanFile) of
+                   true ->
+                       zlib:gunzip(RawData);
+                   false ->
+                       RawData
+                 end,
+          {ok, Meta} = geodata2_format:meta(Data),
+          ets:new(?GEODATA2_STATE_TID, [set, protected, named_table, {read_concurrency, true}]),
+          ets:insert(?GEODATA2_STATE_TID, {data, Data}),
+          ets:insert(?GEODATA2_STATE_TID, {meta, Meta}),
+          {ok, #state{}};
+      _ ->
+          {stop, {geodata2_dbfile_not_found, MagellanFile}}
+    end,
+    DomainState = case filelib:is_file(DomainFile) of
+      true ->
+          {ok, RawData2} = file:read_file(DomainFile),
+          Data2 = case is_compressed(DomainFile) of
+                   true ->
+                       zlib:gunzip(RawData2);
+                   false ->
+                       RawData2
+                 end,
+          {ok, Meta2} = geodata2_format:meta(Data2),
+          ets:new(?GEODATA2_DOMAIN_TID, [set, protected, named_table, {read_concurrency, true}]),
+          ets:insert(?GEODATA2_DOMAIN_TID, {data, Data2}),
+          ets:insert(?GEODATA2_DOMAIN_TID, {meta, Meta2}),
+          {ok, #state{}};
+      _ ->
+          {stop, {ip_to_domain_dbfile_not_found, DomainFile}}
+    end,
+    {ok, {MagellanState, DomainState}}.
 
 -spec init(_) -> {ok, state()} | {stop, tuple()}.
 init(_Args) ->
-    case get_env(geodata2, dbfile) of
-        {ok, File} ->
-            new(File);
-        _ ->
-            {stop, {geodata2_dbfile_unspecified}}
-    end.
+    MagellanFile = case get_env(geodata2, dbfile) of
+      {ok, File} -> File;
+      _ ->
+          {stop, {geodata2_dbfile_unspecified}}
```

Comment:
When a init() function gets as return a {stop, Reason} it means that there is a severe error condition and the application must stop. Here MagellanFile could be {stop, ...) which is passed along to new/2 but what should really happen is to just stop the application as a required configuration is missing. A good way to do this is to nest cases like this. 
```
case get_env(geodata2, dbfile) of
        {ok, MagellanFile} ->	     
              case get_env(ip_to_domain, dbfile) of
                   {ok, DomainFile} ->
                         new(MagellanFile, DomainFile);
                   _ ->
                        {stop, {ip_to_domain_dbfile_unspecified}}
               end,
        _ ->	 
             {stop, {geodata2_dbfile_unspecified}}
end.
```